### PR TITLE
Issue 1568 npm to yarn

### PR DESCRIPTION
**Pull request by Tonours**
**Summary**

Add a yarn.lock file at the root of the project.
Replace NPM with Yarn in travis.yml file.

**Test Plan**

Ensure that travis run without errors.
All current npm scripts has been manually tested (prepublish, pretest, build, dev, postbuild, lint, format, flow, test, test-ci), there are many lint warning on test files (see attached screenshots).

lint-error

- [x] get yarn set up for yourself
- [x] `cd draft-js && rm -rf node_modules && yarn install`
- [x] verify that builds, lint, tests, etc. are passing
- [x] commit the resulting `yarn.lock` file that gets added to the root of the repo
- [x] update `.travis.yml` to use `yarn` instead of `npm` for all commands. Note that the `yarn` commands may vary a bit from the npm CLI commands.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,30 +26,30 @@ addons:
...
+cache:
+  yarn: true
 
 install:
-# Remove any unncessary packages. Probably unnecessary but potentially helpful
-- npm prune
 # Update runs install, but will also pull in the latest updates so the cache
 # doesn't get stale
-- npm update
+- yarn upgrade
```
Comment (zpao) [inline .travis.yml]: Let's not keep this step - it's not relevant with yarn usage and locked dependencies, and in fact might cause problems. It was originally there to catch the case where npm deps would get updates and our dependency ranges would result in breakages that we didn't see. Locks prevent that.

**Comment by mitermayer** [comment]
Hi @Tonours,

Thanks for submitting this PR, if you can amend the comments above from Paul I then should be able to land this.

**Comment by Tonours** [comment]
Hi @zpao and @mitermayer !

Good catch for website package.json and yarn upgrade in travis @zpao 👍 
So, the website now have a yarn.lock file :-)

**Comment by mitermayer** [comment]
@zpao , Now that we will move from npm to yarn, can we do something about this line on our `package.json` ?

```json
"devEngines": {
    ...
    "npm": "2.x || 3.x"
}
```

Can we either remove that `npm` engine or bump it to latest `npm` ?

**Comment by mitermayer** [comment]
Also @Tonours, do you mind rebasing your pull request with master to fix your segfault ?

**Comment by mitermayer** [comment]
Looks like the builds are failing in travis due to that `npm` version on our `package.json`

```
yarn run v1.3.2
$ node node_modules/fbjs-scripts/node/check-dev-engines.js package.json
assert.js:42
  throw new errors.AssertionError({
  ^
AssertionError [ERR_ASSERTION]: Current npm version is not supported for development, expected "5.5.1" to satisfy "2.x || 3.x".
```

**Comment by Tonours** [comment]
Hi @mitermayer !

Thanks for your feedback 👍 
I've updated npm devEngines requirements for npm.
Let me know if everything is good for you.

**Comment by mitermayer** [comment]
I want to merge this since we need a `yarn.lock` to fix our builds, just want to make sure that @zpao  is fine with the changes to `devEnvines` to be pointing to `"npm": "5.x"` if so I will go ahead and merge this

cc @flarnie  @zpao

**Pull request closed without merging**